### ISSUE TITLE
MethodError: no method matching checkbounds(::CSV.PointerString, ::Int64)

### ISSUE THREAD

**Issue by bluesmoon** [JuliaData/CSV.jl#884]
I'm using CSV 0.8.5 on Julia 1.5.4
```
julia> versioninfo()
Julia Version 1.5.4
Commit 69fcb5745b (2021-03-11 19:13 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
```

I have a very simple CSV with just a single line:

```
"can't redefine non-configurable property ""userAgent"""
```

And I try to read it with the following Julia code:

```julia
using CSV, DataFrames

CSV.File("myfile.csv"; header=0) |> DataFrame
```

I immediately get this exception:
```
ERROR: MethodError: no method matching checkbounds(::CSV.PointerString, ::Int64)
Closest candidates are:
  checkbounds(::DataStructures.DiBitVector, ::Integer) at /home/ubuntu/.julia/packages/DataStructures/nBjdy/src/dibit_vector.jl:30
  checkbounds(::AbstractArray, ::Any...) at abstractarray.jl:504
  checkbounds(::AbstractString, ::Union{Integer, AbstractArray}) at strings/basic.jl:214
  ...
Stacktrace:
 [1] codeunit at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/utils.jl:41 [inlined]
 [2] unescape(::CSV.PointerString, ::UInt8) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/utils.jl:338
 [3] columnname(::Array{UInt8,1}, ::Int64, ::Int64, ::Int16, ::Parsers.Options{false,true,true,false,Missing,UInt8,Nothing}, ::Int64) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/detection.jl:259
 [4] readsplitline at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/detection.jl:245 [inlined]
 [5] detectcolumnnames(::Array{UInt8,1}, ::Int64, ::Int64, ::Int64, ::Parsers.Options{false,true,true,false,Missing,UInt8,Nothing}, ::Int64, ::Bool) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/detection.jl:177
 [6] Header at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/header.jl:175 [inlined]
 [7] CSV.File(::String; header::Int64, normalizenames::Bool, datarow::Int64, skipto::Nothing, footerskip::Int64, transpose::Bool, comment::Nothing, use_mmap::Nothing, ignoreemptylines::Bool, select::Nothing, drop::Nothing, missingstrings::Array{String,1}, missingstring::String, delim::Nothing, ignorerepeated::Bool, quotechar::Char, openquotechar::Nothing, closequotechar::Nothing, escapechar::Char, dateformat::Nothing, dateformats::Nothing, decimal::UInt8, truestrings::Array{String,1}, falsestrings::Array{String,1}, type::Nothing, types::Nothing, typemap::Dict{Type,Type}, pool::Float64, lazystrings::Bool, strict::Bool, silencewarnings::Bool, debug::Bool, parsingdebug::Bool, kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/file.jl:223
 [8] top-level scope at REPL[1]:1
```

Any idea what could be wrong and what I could do to fix it? This issue is very reproducible.

**Comment by bluesmoon** [comment]
Update: Removing the leading and trailing `"` from the file fixes the issue, but I still don't understand why it happens.

**Comment by bluesmoon** [comment]
Further update: I can avoid the issue by passing `quotechar='\0'` however this isn't what I want. What I really want is the ability to specify `escapechar='\\'`, unfortunately this gives me a different error:

```
julia> CSV.File("myfile.csv"; header=0, delim=",", escapechar='\\') |> DataFrame
ERROR: ArgumentError: provide a valid sink argument, like `using DataFrames; CSV.read(source, DataFrame)`
Stacktrace:
 [1] read(::String, ::Nothing; copycols::Bool, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/CSV.jl:43
 [2] read(::String, ::Nothing) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/CSV.jl:42 (repeats 2 times)
 [3] getsource(::String) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/utils.jl:262
 [4] Header at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/header.jl:109 [inlined]
 [5] CSV.File(::String; header::Int64, normalizenames::Bool, datarow::Int64, skipto::Nothing, footerskip::Int64, transpose::Bool, comment::Nothing, use_mmap::Nothing, ignoreemptylines::Bool, select::Nothing, drop::Nothing, missingstrings::Array{String,1}, missingstring::String, delim::String, ignorerepeated::Bool, quotechar::Char, openquotechar::Nothing, closequotechar::Nothing, escapechar::Char, dateformat::Nothing, dateformats::Nothing, decimal::UInt8, truestrings::Array{String,1}, falsestrings::Array{String,1}, type::Nothing, types::Nothing, typemap::Dict{Type,Type}, pool::Float64, lazystrings::Bool, strict::Bool, silencewarnings::Bool, debug::Bool, parsingdebug::Bool, kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/file.jl:223
 [6] top-level scope at REPL[1]:1
caused by [exception 1]
SystemError: opening file "myfile.csv": No such file or directory
Stacktrace:
 [1] systemerror(::String, ::Int32; extrainfo::Nothing) at ./error.jl:168
 [2] #systemerror#48 at ./error.jl:167 [inlined]
 [3] systemerror at ./error.jl:167 [inlined]
 [4] open(::String; lock::Bool, read::Bool, write::Nothing, create::Nothing, truncate::Nothing, append::Nothing) at ./iostream.jl:284
 [5] open(::String, ::String; lock::Bool) at ./iostream.jl:346
 [6] open(::String, ::String) at ./iostream.jl:346
 [7] open(::Mmap.var"#7#8"{Array{UInt8,1},Bool,Bool,Tuple{Int64},Int64}, ::String, ::Vararg{String,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:323
 [8] open at ./io.jl:323 [inlined]
 [9] #mmap#6 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Mmap/src/Mmap.jl:243 [inlined]
 [10] mmap at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Mmap/src/Mmap.jl:243 [inlined] (repeats 3 times)
 [11] getsource(::String) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/utils.jl:258
 [12] Header at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/header.jl:109 [inlined]
 [13] CSV.File(::String; header::Int64, normalizenames::Bool, datarow::Int64, skipto::Nothing, footerskip::Int64, transpose::Bool, comment::Nothing, use_mmap::Nothing, ignoreemptylines::Bool, select::Nothing, drop::Nothing, missingstrings::Array{String,1}, missingstring::String, delim::String, ignorerepeated::Bool, quotechar::Char, openquotechar::Nothing, closequotechar::Nothing, escapechar::Char, dateformat::Nothing, dateformats::Nothing, decimal::UInt8, truestrings::Array{String,1}, falsestrings::Array{String,1}, type::Nothing, types::Nothing, typemap::Dict{Type,Type}, pool::Float64, lazystrings::Bool, strict::Bool, silencewarnings::Bool, debug::Bool, parsingdebug::Bool, kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/ubuntu/.julia/packages/CSV/Zl2ww/src/file.jl:223
 [14] top-level scope at REPL[1]:1
```

**Comment by gustafsson** [comment]
I encountered the same error message and managed to find a solution.

It initially seemed to me that it depended on which packages that were loaded, or maybe the order of loading them. It disappeared however after a `Pkg.update()` even though I'm still on CSV 0.8.5 in my project due to other dependencies.

Would you get the same error if you run `Pkg.update()` or if you generate a new project?

```
$ julia
Version 1.6.2 (2021-07-14)

(@v1.6) pkg> generate MyfileCsvTest
  Generating  project MyfileCsvTest:
    MyfileCsvTest/Project.toml
    MyfileCsvTest/src/MyfileCsvTest.jl

(@v1.6) pkg> activate MyfileCsvTest
  Activating environment at `~/MyfileCsvTest/Project.toml`

(MyfileCsvTest) pkg> add CSV, DataFrames
    Updating registry at `~/.julia/registries/General`
    Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Resolving package versions...
    Updating `~/MyfileCsvTest/Project.toml`
  [336ed68f] + CSV v0.9.4
  [a93c6f00] + DataFrames v1.2.2
    Updating `~/MyfileCsvTest/Manifest.toml`
(omitted...)
julia> using CSV, DataFrames

julia> x = CSV.File("myfile.csv"; header=0) |> DataFrame;

julia> show(x, truncate=100)
1×1 DataFrame
 Row │ Column1                                              
     │ String                                               
─────┼──────────────────────────────────────────────────────
   1 │ can't redefine non-configurable property "userAgent"

julia> 
```

This seems to load the file correctly.

**Comment by quinnj** [comment]
Yes, this has been fixed as of the 0.9 release; please update accordingly!
